### ISSUE TITLE
Migrations & Initial Creation

### ISSUE THREAD

**Issue by jamsoft** [PomeloFoundation/Pomelo.EntityFrameworkCore.MySql#1059]
### Steps to reproduce
Create a model in code (Code First)
Add Asp.Net Identity
Enable Migrations
Execute Add-Migration InitialCreate
Execute Update-Database

### The issue
The particular issue I was seeing Was it trying to create AspNetUsers multiple times. Table already exists errors.

I'm also seeing errors in the package manager console immediately after the command, then it attempts to carry on with the command. In the case of Add-Migration is works.

```
Exception message:
Table 'AspNetRoles' already exists ---> MySql.Data.MySqlClient.MySqlException (0x80004005): Table 'AspNetRoles' already exists

Stack trace:
Applying migration '20200407002256_InitialCreate'.
Failed executing DbCommand (10ms) [Parameters=[], CommandType='Text', CommandTimeout='30']
CREATE TABLE `core`.`AspNetRoles` (
    `Id` char(36) NOT NULL,
    `Name` varchar(256) NULL,
    `NormalizedName` varchar(256) NULL,
    `ConcurrencyStamp` longtext NULL,
    CONSTRAINT `PK_AspNetRoles` PRIMARY KEY (`Id`)
);
MySql.Data.MySqlClient.MySqlException (0x80004005): Table 'AspNetRoles' already exists ---> MySql.Data.MySqlClient.MySqlException (0x80004005): Table 'AspNetRoles' already exists
   at MySqlConnector.Core.ResultSet.ReadResultSetHeaderAsync(IOBehavior ioBehavior) in C:\projects\mysqlconnector\src\MySqlConnector\Core\ResultSet.cs:line 49
   at MySql.Data.MySqlClient.MySqlDataReader.ActivateResultSet() in C:\projects\mysqlconnector\src\MySqlConnector\MySql.Data.MySqlClient\MySqlDataReader.cs:line 111
   at MySql.Data.MySqlClient.MySqlDataReader.CreateAsync(CommandListPosition commandListPosition, ICommandPayloadCreator payloadCreator, IDictionary`2 cachedProcedures, IMySqlCommand command, CommandBehavior behavior, IOBehavior ioBehavior, CancellationToken cancellationToken) in C:\projects\mysqlconnector\src\MySqlConnector\MySql.Data.MySqlClient\MySqlDataReader.cs:line 338
   at MySqlConnector.Core.CommandExecutor.ExecuteReaderAsync(IReadOnlyList`1 commands, ICommandPayloadCreator payloadCreator, CommandBehavior behavior, IOBehavior ioBehavior, CancellationToken cancellationToken) in C:\projects\mysqlconnector\src\MySqlConnector\Core\CommandExecutor.cs:line 63
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQueryAsync(IOBehavior ioBehavior, CancellationToken cancellationToken) in C:\projects\mysqlconnector\src\MySqlConnector\MySql.Data.MySqlClient\MySqlCommand.cs:line 227
   at MySql.Data.MySqlClient.MySqlCommand.ExecuteNonQuery() in C:\projects\mysqlconnector\src\MySqlConnector\MySql.Data.MySqlClient\MySqlCommand.cs:line 75
   at Pomelo.EntityFrameworkCore.MySql.Storage.Internal.MySqlConverterCommandBuilderFactory.MySqlConverterRelationalCommandBuilder.MySqlConverterRelationalCommand.Execute(IRelationalConnection connection, DbCommandMethod executeMethod, IReadOnlyDictionary`2 parameterValues)
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.ExecuteNonQuery(IRelationalConnection connection, IReadOnlyDictionary`2 parameterValues)
   at Microsoft.EntityFrameworkCore.Migrations.MigrationCommand.ExecuteNonQuery(IRelationalConnection connection, IReadOnlyDictionary`2 parameterValues)
   at Microsoft.EntityFrameworkCore.Migrations.Internal.MigrationCommandExecutor.ExecuteNonQuery(IEnumerable`1 migrationCommands, IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.Migrations.Internal.Migrator.Migrate(String targetMigration)
   at Microsoft.EntityFrameworkCore.Design.Internal.MigrationsOperations.UpdateDatabase(String targetMigration, String contextType)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.UpdateDatabase.<>c__DisplayClass0_1.<.ctor>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.Execute(Action action)
Table 'AspNetRoles' already exists
```

### Further technical details
My startup code:
services.AddDbContext<CoreDbContext>(options =>
    options.UseMySql(Configuration.GetConnectionString("CoreDbContextConnection"), mysqlOptions =>
        mysqlOptions.ServerVersion(new Version(10, 3, 21), ServerType.MariaDb)
        ));

MySQL version: MariaDB 10
Operating system: Windows  10 Devbox, MariaDB on a Synology
Pomelo.EntityFrameworkCore.MySql version: 2.2.6
Microsoft.AspNetCore.App version: 2.2.0

Other details about my project setup:

**Comment by mguinness** [comment]
See https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/issues/256 for a similar issue.

**Comment by jamsoft** [comment]
I'm creating an entirely new database. It creates the _Migration table just fine. Then complains that a table that doesn't exist in the database cannot be created as it already exists.

Once i had installed the MySQL Nugets and updated the context and connection string I created a new "InitialCreate" Migration and tried to run it which produced this error.

**Comment by jamsoft** [comment]
OK, So this is potentially a bug ... Or a misunderstanding on my part. I had originally been using a SQLServer database for early development before siwtching to MySQL.

I had the following line in my `OnModelCreating` method:

`builder.HasDefaultSchema("core");`

Removing this and everything has started working.

**Comment by lauxjpn** [comment]
This has already been fixed in 3.0.0. See #923 and #982 for further details.
